src/main/java/net/covers1624/curl4j/core/Callback.java

```diff
@@ -158,7 +158,9 @@ protected static long ffi_prep_cif(long rtype, long... atypes) {
 
     // Invoked by JNI in native land, simply forwards the callback through.
     private void ffi_callback(long ret, long args) throws Throwable {
-        assert delegate != null;
+        if (delegate == null) {
```

Comment:
This can just be removed altogether. It's guaranteed to be non-null.

Reply:
I should also clarify, those constructors may seem dead, but they are marked as `protected` specifically to allow sub-classing and overriding of the `read` or `write`, etc methods.

I found this quite useful in other projects when my callback needed a bunch of state and I wanted to isolate it.